zerver/views/home.py

```diff
@@ -159,7 +159,7 @@ def home_real(request):
     if not request.path.startswith(url_lang):
         translation.activate(default_language)
 
-    request.session[translation.LANGUAGE_SESSION_KEY] = default_language
+    request.session[translation.LANGUAGE_SESSION_KEY] = translation.get_language()
```

Comment:
Can you explain this change?  It seems to override the language set in the Zulip UI...

Reply:
(probably just for debugging purposes and forgot to change back)

Reply:
If we use `default_language`, we end up storing 'zh_Hans' in the session, which is locale. I think for languages which don't have '-' character in their name, name is equal to the locale.

Reply:
`translation.get_language()` will return the active language i.e. 'zh-hans'.

Reply:
Well, the point is, that this is the place is the code where the language you picked using the "display settings" modal for language is actually used.  So, if we need to change the case, that's fine, but we probably don't want to ignore that value if it's set?

Reply:
Wait. Let me test this more.

Reply:
So in the 'display settings' modal we use locales ('zh_Hans'). If we pass a locale to `translation.activate()` then Django converts it into a language name using `django.utils.translation.trans_real.to_language` function. I think this is probably why if we store 'zh_Hans' in the session, it will still work. Django uses `django.utils.translation.trans_real.get_language_from_request` to retrieve language from the session and it seems that it expects 'zh-hans' over there.

>but we probably don't want to ignore that value if it's set?

I have moved this statement inside the `if` block. It was working correctly before because the language is activated in the locale middleware but it looked buggy.

Reply:
I see, my confusion was actually that I'd missed the immediately previous line was still there, and thus was confused as to why we're no longer interacting with `default_language` at all here.